### PR TITLE
Update to 2.0.0-beta4

### LINKED ISSUE #1: Update to 2.0.0~beta4?
Perhaps you're interested in updating this to [2.0.0~beta4](https://github.com/ocaml/opam/releases/tag/2.0.0-beta4). I'd love to contribute but my understanding of Ruby and Homebrew are quite limited.

### PR DESCRIPTION
**Pull request by bobbypriambodo**
Also removes aspcud dependency in favor of glpk as build dep to utilize the internal solver.

Closes #1.

### CODE REVIEW

File: Formula/opam@2.rb [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
-  # aspcud has a fairly large buildtime dep tree, and uses gringo,
-  # which requires C++11 and is inconvenient to install pre-10.8
-  if MacOS.version > 10.7
-    depends_on "aspcud" => :recommended
-  else
-    depends_on "aspcud" => :optional
-  end
-
-  needs :cxx11 if build.with? "aspcud"
+  depends_on "glpk" => :build
```
Comment (dbuenzli) [inline Formula/opam@2.rb]: I'm not sure if don't want still want to use `aspcud` by default on platforms that have it. @AltGr @avsm opinions ?

Comment (AltGr) [reply Formula/opam@2.rb]: If opam was compiled with the built-in solver, that will be used by default even if `aspcud` is present. If it was compiled without, `aspcud`, or another solver, will be required at run-time.
Note that master doesn't even need `glpk` anymore, it's vendored in.

Comment (dbuenzli) [reply Formula/opam@2.rb]: Thanks for your answer @AltGr but it doesn't really answer the question. Do you think we should compile and use `opam` with the built-in solver or not on on macos ?

Comment (AltGr) [reply Formula/opam@2.rb]: The built-in solver is much better in normal use-cases, so I think you should prefer it when possible.

Comment (bobbypriambodo) [reply Formula/opam@2.rb]: FWIW the reason I decided to use `glpk` and drop `aspcud` is that the [build instruction of opam](https://github.com/ocaml/opam#compiling-this-repo) seems to require `glpk`, which I understand in turn will [enable the built-in solver](https://github.com/ocaml/opam#integrated-solver), and the whole README seems to me like it is advocating the use of the internal solver (might be wrong though).

We could have some logic to choose either one, but if the built-in one is recommended then I think it's much simpler this way. Would love your input.

Comment (dbuenzli) [reply Formula/opam@2.rb]: Thanks @AltGr ! @bobbypriambodo could you please update the PR accordingly.

Comment (bobbypriambodo) [reply Formula/opam@2.rb]: @dbuenzli I'm not sure I understand what you mean, what should I update?

**Comment by dbuenzli** [comment]
Ah sorry I got confused by that: 

> Note that master doesn't even need glpk anymore, it's vendored in.

**Comment by bobbypriambodo** [comment]
Thanks @dbuenzli!

**Comment by mars0i** [comment]
Thanks to everyone involved!

**Pull request merged**